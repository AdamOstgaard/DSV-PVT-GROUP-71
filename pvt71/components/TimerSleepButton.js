import React from "react";
import { TouchableOpacity, StyleSheet, Image, Alert, Button } from "react-native";

export default class TimerSleepButton extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      enabled: false
    };
    this.showConfirmWindow = this.showConfirmWindow.bind(this);
  }
  componentDidUpdate(prevProps, prevState) {
    if (prevProps.onResume !== this.props.onResume) {
      this.setState({ enabled: this.props.onResume });
    }
  }
  render() {
    return (
      <TouchableOpacity style={styles.button} onPress={this.showConfirmWindow}>
        <Image source={require("../assets/images/zzz-icon-3.png")} />
      </TouchableOpacity>
    );
  }

  toggle() {
    const val = !this.state.enabled;
    this.props.onToggle(val);
    this.setState({ enabled: val });
  }

  showConfirmWindow() {
    if (!this.state.enabled) {
      Alert.alert(
        "Sov",
        "Pausa timern till nÃ¤sta starttid?",
        [
          { text: "Ja", onPress: () => this.toggle() },
          {
            text: "Avbryt",
            onPress: () => console.log("Avbryt"),
            style: "cancel"
          }
        ],
        { cancelable: false }
      );
    }
  }
}

const styles = StyleSheet.create({
  button: {
    height: 90,
    width: 90
  }
});
