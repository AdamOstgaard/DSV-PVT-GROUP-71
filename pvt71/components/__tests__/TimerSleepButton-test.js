import "react-native";
import React from "react";
import TimerSleepButton from "../TimerSleepButton";
import renderer from "react-test-renderer";
import Alert from "react-native";
import { create } from "react-test-renderer";

describe("Timer Test", () => {
  describe("Timer Snapshot", () => {
    it("Renders correctly", () => {
      const tree = renderer.create(<TimerSleepButton />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("test toggle()", () => {
    it("should update state variable enable to true", () => {
      let button = renderer
        .create(
          <TimerSleepButton
            onToggle={enabled => console.log(enabled)}
            onResume={resume => console.log(resume)}
          />
        )
        .getInstance();
      button.toggle();
      expect(button.state.enabled).toEqual(true);
    });
  });

 // describe("test componentDidUpdate()", () => {
  //  it("should not update variable enable to true", () => {
  //    let button = renderer
  //      .create(<TimerSleepButton onToggle={true} onResume={true} />)
  //      .getInstance();
  //    button.componentDidUpdate(button.props.onResume);
  //    expect(button.state.enabled).toEqual(false);
  //  });
 // });
});
