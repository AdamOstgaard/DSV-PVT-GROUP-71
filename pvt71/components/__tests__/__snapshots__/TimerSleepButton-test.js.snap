// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer Test Timer Snapshot Renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 90,
      "opacity": 1,
      "width": 90,
    }
  }
>
  <Image
    source={1}
  />
</View>
`;
